fleetdm/fleet
# Orbit: Add early update checks before starting sub-systems

#5875

- [X] Changes file added for user-visible changes (in `changes/` and/or `orbit/changes/`).
~- [ ] Documented any API changes (docs/Using-Fleet/REST-API.md)~
~- [ ] Documented any permissions changes~
~- [ ] Ensured that input data is properly validated, SQL injection is prevented (using placeholders for values in statements)~
~- [ ] Added support on fleet's osquery simulator `cmd/osquery-perf` for new osquery data ingestion features.~

QA: I've tested the changes by:
1. Running local TUF server and generate packages (via `tools/tuf/test/main.sh` script).
2. Install packages on Ubuntu, Windows and macOS.
3. Smoke test installation.
4. Upgrade and publish Orbit (via `tools/tuf/test/push_target.sh`).
5. Check successful update on the three hosts.
6. Stop Orbit service on the three hosts.
7. Upgrade and publish Orbit (via `tools/tuf/test/push_target.sh`).
8. Start Orbit service on the three hosts.
9. Check successful update on the three hosts.
